tests/struct.rs

```diff
@@ -0,0 +1,99 @@
+use serde::{Deserialize, Serialize};
+use serde_bin_prot::to_writer;
+use std::f64::INFINITY;
+
+mod common;
+use common::TestCase;
+
+#[derive(Debug)]
+enum StructTestCases {
+    TestA(A),
+    TestB(B),
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct A {
+    x: i64,
+    y: f64,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct B {
+    y: b_inner,
+    z: Option<i64>,
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct b_inner {
+    w: i64,
+    x: i64,
+}
+
+fn struct_test_cases() -> Vec<TestCase<StructTestCases>> {
```

Comment:
I'm not sure if there's a cleaner way to create all the test cases in this function, let me know if there is a nicer way

Reply:
cool, I like the idea of asserting to a byte array instead of a string, it seems cleaner that having to format a string even if we need to do some manual 0x adding. and I just meant the cases for `struct` in this file

Reply:
We could also write a macro that takes the Jane St style byte arrays and produces a  `Vec<u8>`. I think there is a lot of value to be had from leveraging their test suite as much as possible.